src/pages/library/datasetBuilder/CohortEditor.test.ts

```diff
@@ -57,10 +72,51 @@ describe('CohortEditor', () => {
       kind: 'list',
       values: [{ id: 0, name: 'value' }],
     });
-    render(createCriteriaViewComponent(_.noop)(criteria));
+    render(
+      h(CriteriaView, {
+        deleteCriteria: _.noop,
+        updateCriteria: _.noop,
+        criteria,
+        key: criteria.id,
+      })
+    );
 
     expect(screen.getByText(criteria.name, { exact: false })).toBeTruthy();
-    expect(screen.getByText(criteria.value.name)).toBeTruthy();
+    expect(screen.getByText(criteria.valuesSelected[0].name, { exact: false })).toBeTruthy();
+  });
+
+  it('updates when list updated', async () => {
+    // Arrange
+    const user = userEvent.setup();
+    const criteria = criteriaFromOption({
+      id: 0,
+      name: 'list',
+      kind: 'list',
+      values: [{ id: 0, name: 'value' }],
+    });
+    let newCriteria = criteriaFromOption({
+      id: 0,
+      name: 'list',
+      kind: 'list',
+      values: [{ id: 0, name: 'value' }],
+    });
+    render(
+      h(CriteriaView, {
+        deleteCriteria: _.noop,
+        updateCriteria: (updatedCriteria) => {
+          newCriteria = updatedCriteria as ProgramDataListCriteria;
+        },
+        criteria,
+        key: criteria.id,
+      })
+    );
+    // Act
+    const select = screen.getByLabelText('Select one or more list');
+    await user.click(select);
+    const valueOption = await screen.findAllByText('value');
+    await user.click(valueOption[0]);
+    // Assert
+    expect(newCriteria.valuesSelected.length).toBe(0);
```

Comment:
I'm not sure what this is asserting. Is the effect that the number of values selected go from 1 to 0? If so, could the code assert the 1 state before asserting the 0 state?

If you instead mock `updateCriteria` you could assert on the parameter to `updateCriteria` here which I think is more direct than asserting on state that `updateCriteria` writes to.